packages/noya-state/src/selectors/elementSelectors.ts

```diff
@@ -21,12 +21,30 @@ type PathElement =
       value: SelectedControlPoint;
     };
 
+export function canClosePath(state: ApplicationState, element: PathElement) {
+  if (element.type === 'point') {
+    const [layerId, index] = element.value;
+
+    const layers = getSelectedLayers(state)
```

Comment:
This should work, but rather than looking up an array of layers, filtering it down, and picking the 0 element, we can find a layer directly and more safely via:

```ts
const layer = Layers.find(getCurrentPage(state), (layer) => layer.do_objectID === layerId);

if (!layer || !Layers.isPointsLayer(layer)) return false
```

If we do this in maybe 1 more place, we can make a `findLayerInCurrentPage` util